src/sentry/flags/providers.py

```diff
@@ -238,6 +241,122 @@ def validate(self, message_bytes: bytes) -> bool:
+    createdBy = serializers.CharField(required=True)
+    createdByUserId = serializers.IntegerField(required=False, allow_null=True)
+    type = serializers.CharField(allow_blank=True, required=True)
+    tags = serializers.ListField(
+        child=serializers.DictField(child=serializers.CharField()), required=False, allow_null=True
+    )
+    project = serializers.CharField(required=False, allow_null=True)
+    environment = serializers.CharField(required=False, allow_null=True)
+
+
+def _get_user(validated_event: dict[str, Any]) -> tuple[str, int]:
+    """If the email is not valid, default to the user ID sent by Unleash."""
+    created_by = validated_event["createdBy"]
+    try:
+        serializers.EmailField().run_validation(created_by)
+        return created_by, CREATED_BY_TYPE_MAP["email"]
+    except ValidationError:
+        pass
+
+    if "createdByUserId" in validated_event:
+        return validated_event["createdByUserId"], CREATED_BY_TYPE_MAP["id"]
+    return created_by, CREATED_BY_TYPE_MAP["name"]
+
+
+class UnleashProvider:
+    provider_name = "unleash"
+
+    def __init__(self, organization_id: int, signature: str | None) -> None:
+        self.organization_id = organization_id
+        self.signature = signature
+
+    def handle(self, message: dict[str, Any]) -> list[FlagAuditLogRow]:
+        serializer = UnleashItemSerializer(data=message)
+        if not serializer.is_valid():
+            raise DeserializationError(serializer.errors)
+
+        result = serializer.validated_data
+        action = result["type"]
+
+        if action not in SUPPORTED_UNLEASH_ACTIONS:
+            return []
+
+        created_by, created_by_type = _get_user(result)
+        unleash_tags = result.get("tags") or []
+        tags = {tag["type"]: tag["value"] for tag in unleash_tags}
+
+        if result.get("project"):
+            tags["project"] = result.get("project")
+
+        if result.get("environment"):
+            tags["environment"] = result.get("environment")
+
+        return [
+            {
+                "action": _handle_unleash_actions(action),
+                "created_at": result["createdAt"],
+                "created_by": created_by,
+                "created_by_type": created_by_type,
+                "flag": result["featureName"],
+                "organization_id": self.organization_id,
+                "tags": tags,
+            }
+        ]
+
+    def validate(self, message_bytes: bytes) -> bool:
+        # Since the user is passing in whatever they want as an authorization string,
+        # we don't need to validate it like a normal secret.
+        def _dummy_validate_secret(secret: str, message_bytes: bytes) -> str:
+            return secret
+
+        validator = SecretValidator(
+            self.organization_id,
+            self.provider_name,
+            message_bytes,
+            self.signature,
+            secret_validator=_dummy_validate_secret,
+        )
+        return validator.validate()
```

Comment:
You can simplify this section by re-implementing the validation logic for this use case.

```python
if self.signature is None:
    return False

for secret in _query_signing_secrets(...):
    if secret == self.signature:
        return True
return False
```

Reply:
Do you mind renaming `SecretValidator` to `PayloadSignatureValidator`?  The original name is overly broad now that we accept multiple types of secret validation.

Reply:
From my first comment, you could put the suggested logic into a new class called `AuthTokenValidator` that could be re-used for future providers.

Reply:
updated!